internal/nix/profile.go

```diff
@@ -264,6 +261,32 @@ func ProfileInstall(args *ProfileInstallArgs) error {
 	return nil
 }
 
+// ProfileRemoveItems removes the items from the profile, in a single call, using their indexes.
+// It is up to the caller to ensure that the underlying profile has not changed since the items
+// were queried.
+func ProfileRemoveItems(profilePath string, items []*NixProfileListItem) error {
+	if items == nil {
+		return nil
+	}
+
+	indexes := []string{}
+	for _, item := range items {
+		indexes = append(indexes, strconv.Itoa(item.index))
+	}
+	cmd := exec.Command("nix", "profile", "remove",
+		"--profile", profilePath,
+		"--impure", // for NIXPKGS_ALLOW_UNFREE
+		strings.Join(indexes, " "),
```

Comment:
Does this behave OK when multiple indexes? I think we can also remove by unlocked reference but that may be a problem with duplicates